mu-plugins/central-hub/tests/phpunit/integration/meta-data/saveMetaBoxes.php

```diff
@@ -0,0 +1,374 @@
+ * @since       1.3.0
+ * @author      Robert Gadon <rgadon107>
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Integration\Metadata;
+
+use function KnowTheCode\ConfigStore\loadConfig;
+use function spiralWebDB\Metadata\save_meta_boxes;
+use function wp_create_nonce;
+use function add_post_meta;
+use function get_post_meta;
+use function delete_post_meta;
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+
+/**
+ * Class Tests_SaveMetaBoxes
+ *
+ * @package spiralWebDb\centralHub\Tests\Integration\Metadata
+ * @group   meta-data
+ */
+class Tests_SaveMetaBoxes extends Test_Case {
+
+	/**
+	 * Instance of the post object for each test.
+	 *
+	 * @var WP_Post
+	 */
+	protected $post;
+
+	/**
+	 * Empty the store before starting these tests.
+	 */
+	public static function setUpBeforeClass() {
+		self::empty_the_store();
+	}
+
+	/**
+	 * Set up each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		// Create and get the post_id via the factory method.
+		$this->post = self::factory()->post->create();
+	}
+
+	/**
+	 * Cleans up the test environment after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+		// Clean up.
+		self::empty_the_store();
+		$_POST = [];
+		delete_post_meta( $this->post, 'role' );
+		delete_post_meta( $this->post, 'residence_city' );
+		delete_post_meta( $this->post, 'residence_state' );
+		delete_post_meta( $this->post, 'tour_number' );
+	}
+
+	/*
+	* Test save_meta_boxes() should register to add_action( save_post' ) when event fires.
+	*/
+	public function test_function_should_register_to_action_hook_when_event_fires() {
+		$this->assertTrue( has_action( 'save_post' ) );
+		// `has_action()` returns the priority level of the callback registered to the event ($tag).
+		$this->assertEquals( 10, has_action( 'save_post', 'spiralWebDB\Metadata\save_meta_boxes' ) );
+	}
+
+	/**
+	 * Test save_meta_boxes() should not save when custom fields config (config store) and
+	 *    post meta values in $_POST are empty.
+	 */
+	public function test_should_not_save_when_custom_fields_config_and_post_meta_in_POST_are_empty() {
+		// By default, $config is an empty array when the config store is empty.
+		// Add post meta to the database.
+		add_post_meta( $this->post, 'role', 'Bass/Baritone' );
+		add_post_meta( $this->post, 'residence_city', 'Houston' );
+		add_post_meta( $this->post, 'residence_state', 'TX' );
+		add_post_meta( $this->post, 'tour_number', '1' );
+		$_POST = [
+			'post_ID'            => $this->post,
+			'post_status'        => 'publish',
+			'members'            => [
+				'role'            => '',
+				'residence_city'  => '',
+				'residence_state' => '',
+				'tour_number'     => '',
+			],
+			'members_nonce_name' => wp_create_nonce( 'members_nonce_action' ),
+		];
+		$this->assertSame( 'Bass/Baritone', get_post_meta( $this->post, 'role', true ) );
```

Comment:
@rgadon107 please add an empty line between the set up code and the assertions. Why? It helps with readability to see where things are set up vs. where the actual test assertions start.

Please make this change for all of the methods in this file. Thanks.

Reply:
Done.